### ISSUE TITLE
Add new rule `link-types` 

### ISSUE THREAD

**Issue by kagankan** [markuplint/markuplint#719]
# Context

I would like to detect non-existing `rel` attribute value (typo)

```html
<link rel="ikon" href="/icon.png">
<!-- link type `ikon` does not exist, but is not warned  -->
```

Link types are defined in the HTML spec.
https://html.spec.whatwg.org/multipage/links.html#linkTypes

However, the value of the link types can be extended by microformats.
https://html.spec.whatwg.org/multipage/links.html#other-link-types
Therefore, it accepts any string in the spec.

I would like to be able to configure whether microformats are allowed.

# Proposal

Add new rule `link-types`.

```ts
interface {
    "link-types": {
        options: {
            allowMicroformats: boolean | string[]
        }
    }
}
```

### Allow any string

```js
{
    "link-types": {
        options: {
            allowMicroformats: true
        }
    }
}
```

```html
<!-- All correct -->
<link rel="icon" href="/icon.png">
<link rel="apple-touch-icon" href="/icon.png">
<link rel="ikon" href="/icon.png">
```

### Allow only specific link types

```js
{
    "link-types": {
        options: {
            allowMicroformats: ["apple-touch-icon"]
        }
    }
}
```

```html
<!-- Correct: defined in the HTML spec -->
<link rel="icon" href="/icon.png">
<!-- Correct: allowed by the rule's option -->
<link rel="apple-touch-icon" href="/icon.png">
<!-- Incorrect -->
<link rel="ikon" href="/icon.png">
```
